### PR TITLE
Launchpad: Enable use storybook on launchpad package

### LINKED ISSUE #87192: Launchpad: Add Storybook to the launchpad package
### What

Enable to easily see and document the launchpad component and its variations and usages.  

### Why

Launchpad is important on the site creation user journey and we have multiple usages and tons of task variations ( E.g customer-home, tailored onboarding, etc...), so we need to have an easy way to render the launchpad based on the variations to speed up the development and testing process. 

### How
Currently, we have some Storybooks under some packages directory (e.g., [tour-kit](https://github.com/Automattic/wp-calypso/blob/1afe45179cfa7ac98bcdb30dd4d739cb4589f019/packages/tour-kit/.storybook/main.js), [components](https://github.com/Automattic/wp-calypso/blob/1afe45179cfa7ac98bcdb30dd4d739cb4589f019/packages/components/.storybook/main.js#)), but no Storybook for under the launchpad package.

### Context
(copied from #75167) 

Without Storybook, we need to run a local server when developing UI-related things, which is slow to debug and not easy to test all the conditions.

I would be happy to be able to develop based on Storybook at a larger granularity (e.g., per page) without limiting Storybook to so-called "components" only. https://storybook.js.org/docs/react/get-started/why-storybook#build-uis-in-isolation

There are several advantages to using Storybook for UI development.

- Rapid development: Since components are isolated, developers can quickly prototype and iterate on their designs without waiting for a full build and refresh cycle, which can be time-consuming with a local server. 
- Organizing UI: Complex conditions can be organized and developed by component, which is helpful in finding, understanding, testing, and debugging UI patterns
- Storybook makes component testing easy with its play function, addon, etc. https://storybook.js.org/docs/react/writing-tests/introduction

### PR DESCRIPTION
**Pull request by gabrielcaires**
Related to #87192
More context here paYKcK-45j-p2

## Proposed Changes
* Install the storybook to render the components in an isolated context
* Install MSW to mock requests
* Create initial stories for the components: (Launchpad, LaunchpadInternal, Checklist, ChecklistItem. 
* Enable the accessibility plugin

## Testing Instructions
* Run yarn install after the clone
* Run `yarn workspace @automattic/launchpad run storybook`
* Play with the components 
* Go to the launchpad internal component
* Select the "link-in-bio" option as I did on the video.

https://github.com/Automattic/wp-calypso/assets/38718/3114cb18-7b21-4b5c-8b25-936619a4de63

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by gabrielcaires** [comment]
I am investigating an unexpected lint issue outside the scope of my changes,

File: packages/launchpad/static/mockServiceWorker.js [hunk 1]
```diff
@@ -0,0 +1,287 @@
+/* eslint-disable */
+/* tslint:disable */
```
Comment (gabrielcaires) [inline packages/launchpad/static/mockServiceWorker.js:2]: This whole file is generated by MSW, we should not update but we need to track

**Comment by andres-blanco** [comment]
Hi @gabrielcaires ! There's a lot of code here to review, and I am not sure of what is the intent here. Is there some context on what we are trying to achieve with this storybook? That will set me up better for testing.

**Comment by gabrielcaires** [comment]
@andres-blanco as I explained on the issue and the paYKcK-45j-p2 storybook is a very important frontend tool to enable use to work in components separately.  It helps us to shape them easily and better.

Another useful like is:  https://storybook.js.org/docs/get-started/why-storybook

**Comment by markbiek** [comment]
When I run `yarn workspace @automattic/launchpad run storybook`, Storybook launches but I get this error when I click on the `LaunchpadInternal` component.

CleanShot 2024-02-19 at 17 06 17

I also get these errors on the command-line:

CleanShot 2024-02-19 at 17 07 35

**Comment by markbiek** [comment]
I also noticed this among the messages that Storybook puts out:

image

Not sure if this is something we need to be concerned about or not.

**Comment by gabrielcaires** [comment]
@markbiek Oh it missing to run `yarn install ' after checking this branch. I thought we had something automatic to warn us about it.  Testing instructions updated. 

Regarding the second screenshot I will check this telemetry.

**Review by markbiek — APPROVED** [review]
Working as expected!

**Comment by markbiek** [comment]
@gabrielcaires Good find on disabling that telemetry!

**Comment by markbiek** [comment]
@gabrielcaires I'm not sure what the cause of that test failure is. It doesn't seem related to this PR so maybe it's worth rebasing `trunk`?

**Pull request merged**